packages/remix-dev/config.ts

```diff
@@ -353,12 +353,16 @@ export async function readConfig(
       path.join("public", "build")
   );
 
-  if (typeof appConfig.devServerPort !== "number") {
-    appConfig.devServerPort = Number(
-      process.env.REMIX_DEV_SERVER_WS_PORT || 8002
-    );
+  let devServerPort: number;
+
+  // check if the port is already in use as this file is reloaded after a dep install or config change
+  if (process.env.REMIX_DEV_SERVER_WS_PORT) {
+    devServerPort =
+      Number(process.env.REMIX_DEV_SERVER_WS_PORT) ||
+      (await getPort({ port: appConfig.devServerPort }));
+  } else {
+    devServerPort = await getPort({ port: appConfig.devServerPort });
```

Comment:
How about this:

```suggestion
  let devServerPort = Number(process.env.REMIX_DEV_SERVER_WS_PORT) || await getPort({ port: Number(appConfig.devServerPort) || null });
```

Only thing is it wouldn't work with a port of `0`, but I don't think that's a concern... We can't use `??` because `NaN`.

This also handles the case of `devServerPort` not being configured properly.